### ISSUE TITLE
Switch off auto language selection

### ISSUE THREAD

**Issue by HarmHoog** [NamelessMC/Nameless#2861]
### Describe the feature

Currently the auto language selection seems to be enabled for all users. However I have a site in English and it looks weird if half of it is in another language. It would therefor be nice if the auto language selection could be turned off.

**Comment by Browsit** [comment]
See #2826

**Comment by Derkades** [comment]
Isn't that issue about per-user settings instead of a global switch to disable the feature entirely?

**Comment by HarmHoog** [comment]
Yes indeed, I'd prefer to have the feature disabled at all instead of a language selection.

**Comment by Browsit** [comment]
Generally, the decision on whether a site "looks weird" alongside English content is left up to the visitor. While disabling auto-language globally could be useful for other reasons, we suggest the linked issue is a better solution here.

**Comment by HarmHoog** [comment]
I don't agree with that. Imo a site that's just half translated automatically looks unprofessional. Especially if the rest of the content on the site is all in English so there just is no reason for a non-English reader to use the site. Also, the feature of disabling the function at all could perfectly exist next to the suggestion in the linked issue, whether one is a better solution doesn't really matter. It's up to the site's owner to decide which option suits best.

**Comment by PikaMug** [comment]
> Imo a site that's just half translated automatically looks unprofessional. Especially if the rest of the content on the site is all in English so there just is no reason for a non-English reader to use the site.

It might look that way to monolinguists, but this is commonplace for most sites with user-generated content (Amazon, Discord, Google, Meta, Twitter, et al.). If visitors want, say, a forum post translated, there are extensions/apps for that. As for site elements, if you're using hard-coded English strings then sorry but you're doing it wrong. Unless maybe the goal is to only serve English readers but I'm not getting that impression here.

> Also, the feature of disabling the function at all could perfectly exist next to the suggestion in the linked issue, whether one is a better solution doesn't really matter. It's up to the site's owner to decide which option suits best.

Sure it could, but that wasn't the argument. You said the reason is that it looks weird, but current practices would indicate that most non-English readers don't think so or don't mind. Having this functionality available to them is not currently up to the site owner and modern standards would say it's better that way. I'm not against adding this feature but the point is that you shouldn't disable it because you think it looks weird; it doesn't.

**Comment by HarmHoog** [comment]
I get that looking weird is maybe not the best argument but let me clarify a bit further.

I did mean it for a site to only serve English readers. The problem is that not all content on a site is user-generated. For example, we use a static homepage which isn't translated in other languages. Apart from the auto-translated parts, everything is in English. In the case of sites like Discord, Google, Meta etc everything the site itself provides is translated. You will not see the Discord homepage with just half of their content translated, which would be the case for our site which I think looks unprofessional.  Of course we could translate the site into every language but we simply don't have the means for that nor the need since we just can't serve non-English speakers.

For the user-generated content, which in our case would only exist in the Forums, I'd also prefer to not have the content translated. We use the site for support which we only can give in English. Having the site translated to someone's native language suggests that using the forums in another language than English is supported. For instance: I'm Dutch and seeing the forums in Dutch makes me think that I can ask for support in Dutch.

I get that it might be a help to some readers but in the case of our site the problem is that if you can't read English there just isn't anything there for you. If you do want to read the site you can always use those extensions which should be good enough to translate the parts that nameless does anyway.

**Comment by Derkades** [comment]
Site owners should at least be able to disable it by default
